src/Http/Routing/src/Builder/RoutingEndpointConventionBuilderExtensions.cs

```diff
@@ -22,6 +23,12 @@ public static class RoutingEndpointConventionBuilderExtensions
     /// An empty collection means any host will be accepted.
     /// </param>
     /// <returns>A reference to this instance after the operation has completed.</returns>
+    /// <remarks>
+    /// APIs that rely on the <see href="https://developer.mozilla.org/docs/Web/HTTP/Headers/Host">Host header</see>, such as
+    /// <see cref="HttpRequest.Host"/> and <see cref="RequireHost"/>, are vulnerable to potential spoofing by clients.
+    ///
+    /// To prevent host and port spoofing, use <see href="https://learn.microsoft.com/dotnet/api/microsoft.aspnetcore.connections.features.itlshandshakefeature.hostname">ITlsHandshakeFeature.HostName</see> to verify the server name used during the TLS handshake, and <see cref="HttpContext.Connection"/> (<see cref="ConnectionInfo.LocalPort"/>) to verify the local port the connection was accepted on.
```

Comment:
```suggestion
    /// <remarks>
    /// APIs that depend on the <see href="https://developer.mozilla.org/docs/Web/HTTP/Headers/Host">Host header</see>, including
    /// <see cref="HttpRequest.Host"/> and <see cref="RequireHost"/>, are vulnerable to potential client spoofing by clients.
    ///
     /// To safeguard against host and port spoofing, use <see href="https://learn.microsoft.com/dotnet/api/microsoft.spnetcore.connections.features.itlshandshakefeature.hostname">ITlsHandshakeFeature.HostName</see> to confirm the server name used during the TLS handshake, and <see cref="HttpContext.Connection"/> (<see ref="ConnectionInfo.LocalPort"/>) to confirm the local port that accepted the connection.
     /// </remarks>
```

Reply:
@tdykstra  here's what coPilot suggests.
```
/// <remarks>
/// APIs relying on the <see href="https://developer.mozilla.org/docs/Web/HTTP/Headers/Host">Host header</see> (like <see cref="HttpRequest.Host"/> and <see cref="RequireHost"/>) are vulnerable to spoofing attacks. 
///
/// To mitigate host/port spoofing:
/// * Verify the server name used during the TLS handshake using <see href="https://learn.microsoft.com/dotnet/api/microsoft.aspnetcore.connections.features.itlshandshakefeature.hostname">ITlsHandshakeFeature.HostName</see>.
/// * Verify the local port where the connection was accepted using <see cref="HttpContext.Connection"/> (specifically <see cref="ConnectionInfo.LocalPort"/>).
/// </remarks>
```

Reply:
The text sounds good, but the asterisks won't render as list items will they?

Reply:
>  I don't know why the next suggestion doesn't have rich diff-ing. @tdykstra do you know why?

Maybe because it's making a change to text that has a change already waiting to be accepted.

Reply:
> The text sounds good, but the asterisks won't render as list items will they?

I think you're right. @gewarren probably knows.

Reply:
No, you'd need to do something like this: https://github.com/dotnet/runtime/blob/d14ac30978a5894de99d7b3934199d05d239e691/src/libraries/System.Linq/src/System/Linq/Iterator.cs#L16-L29.